### PR TITLE
feat: add static Response.json

### LINKED ISSUE #1669: Impl "Response.json"
**Reproduction**

1. `new Response.json({ foo: 'bar' })`

### PR DESCRIPTION
**Pull request by KhafraDev**
## Purpose

Implements static `Response.json` from https://github.com/whatwg/fetch/pull/1392

## Changes

Adds static Response.json

## Additional information

___

- [ ] I updated readme
- [x] I added unit test(s)

___

- Fixes #1669

### CODE REVIEW

File: src/response.js [hunk 1]
```diff
@@ -124,6 +124,25 @@ export default class Response extends Body {
 		return response;
 	}
 
+	static json(data = undefined, init = {}) {
```
Comment (LinusU) [inline src/response.js:127]: I think that using `init = {}` and assuming `init` is an object further down in the function will make the following code crash:

```js
Response.json({ hej: 1 }, null)
```

But in Chrome that works:

Screenshot 2022-11-09 at 09 22 54

File: src/response.js [hunk 2]
```diff
@@ -124,6 +124,25 @@ export default class Response extends Body {
...
+		}
+
+		return new Response(body, {
+			...init,
+			headers: {
+				// If body’s type is non-null and response’s header list does not contain
+				// `Content-Type`, then append (`Content-Type`, body’s type) to response’s
+				// header list.
+				'Content-Type': 'application/json',
+				...init.headers
```
Comment (LinusU) [inline src/response.js:141]: What happens if `init.headers` contains `content-type` or `cOnTeNt-TyPe`?

File: src/response.js (hunk 1 above)
Comment (LinusU) [inline src/response.js:127]: Not sure if this is in the spec, but Chrome throws different errors if called without `data` vs. explicitly passing `undefined`:

Screenshot 2022-11-09 at 09 29 56

Not sure we need to follow this though, but I think that the `= undefined` should be removed? It communicates that the argument is optional, but it is in fact required...

```suggestion
	static json(data, init = {}) {
```

Comment (KhafraDev) [reply src/response.js:127]: `any` webidl types mean that the type is optional and Response.json.length should be 0

File: src/response.js (hunk 2 above)
Comment (KhafraDev) [reply src/response.js:141]: I have a test for this

File: src/response.js (hunk 1 above)
Comment (KhafraDev) [reply src/response.js:127]: node-fetch doesn't seem to do much argument validation, I followed other areas of the code.

For example when constructing a new response:
```js
new Response('', null)
```

throws an error when it shouldn't

```js
new Response('', 3)
```
works when it shouldn't (same with booleans, etc)

Comment (KhafraDev) [reply src/response.js:127]: see https://github.com/nodejs/undici/pull/1723

**Comment by jimmywarting** [comment]
I'm okey with some small deviations, we don't have all the webidl stuff undici have and don't follow the spec at 100%. it was set up to be "good enough" and solve the most common use cases.
We didn't have all the web parts in NodeJS that builds up fetch for what it is today in the beginning when node-fetch was first built, like web streams, blob, formdata, AbortController, etc. so we did compromise a little bit. 

it's going to be replaced by NodeJS in the newer versions anyway at some point. node-fetch is just a polyfill at this stage. 
it at least gets the job done and i'm sure ppl will get it right 99% of the times.

File: src/response.js (hunk 1 above)
Comment (LinusU) [reply src/response.js:127]: That's cool, today I learned!

**Review by LinusU — APPROVED** [review]
Looks great! 👍

**Review by jimmywarting — APPROVED** [review]

**Pull request merged**